src/test/kotlin/org/ionproject/integration/file/PDFFileDownloaderTest.kt

```diff
@@ -0,0 +1,90 @@
+
+
+package org.ionproject.integration.file
+import java.net.ConnectException
+import java.nio.file.FileSystems
+import java.nio.file.Path
+import java.nio.file.PathMatcher
+import java.nio.file.Paths
+import org.ionproject.integration.file.`interface`.FileDownloader
+import org.ionproject.integration.file.exception.InvalidFormatException
+import org.ionproject.integration.file.exception.ServerErrorException
+import org.ionproject.integration.file.implementation.PDFFileDownloader
+import org.ionproject.integration.utils.Try
+import org.ionproject.integration.utils.orThrow
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertFalse
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertDoesNotThrow
+import org.junit.jupiter.api.assertThrows
+internal class PDFFileDownloaderTest {
+    companion object {
+        private val pdfDownloader: FileDownloader = PDFFileDownloader()
+        inline fun <reified T : Throwable> downloadAndAssertThrows(url: String, dstFile: String) {
+            assertThrows<T> { downloadPdf(url, dstFile).orThrow() }
+        }
+        fun downloadPdf(url: String, dstFile: String): Try<Path> {
+            return pdfDownloader.download(url, dstFile)
+        }
+        fun deleteFile(path: Path) {
+            path.toFile().delete()
+        }
+        fun assertFileDoesntExist(path: String) {
+            val file = Paths.get(path).toFile()
+            assertFalse(file.exists())
+        }
+    }
+    @Test
+    fun whenValid_ThenDownloadIsSuccessful() {
+        val url = "https://www.w3.org/WAI/ER/tests/xhtml/testfiles/resources/pdf/dummy.pdf"
+        val fileDst = "/tmp/dummy.pdf"
+        val matcher: PathMatcher = FileSystems.getDefault().getPathMatcher("glob:**.pdf")
+        val path = assertDoesNotThrow { downloadPdf(url, fileDst).orThrow() }
+        assertTrue(matcher.matches(path))
+        assertEquals(String(path.toFile().readBytes().slice(0..6).toByteArray()), "%PDF-1.")
```

Comment:
If the file exists that is has to be an .pdf file since there a checkFormat function.
Maybe there could be explicit tests to this checkFormat function

Reply:
Good point!